File: app/src/main/java/com/sjani/usnationalparkguide/UI/Details/Campgrounds/CampDetailViewModel.java
package com.sjani.usnationalparkguide.UI.Details.Campgrounds;


import com.sjani.usnationalparkguide.Data.CampEntity;
import com.sjani.usnationalparkguide.Utils.ParkRepository;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

class CampDetailViewModel extends ViewModel {

    private final ParkRepository repository;
    private final String apiKey;
    private final String parkCode;
    private LiveData<CampEntity> camp;

    public CampDetailViewModel(ParkRepository repository, String apiKey, String parkCode) {
        this.repository = repository;
        this.apiKey = apiKey;
        this.parkCode = parkCode;
    }

    public LiveData<CampEntity> getCamp(String campId) {
        camp = this.repository.getCampFromDb(campId);
        return camp;
    }

}

File: app/src/main/java/com/sjani/usnationalparkguide/ui/viewmodel/CampDetailViewModel.kt
package com.sjani.usnationalparkguide.ui.viewmodel

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import androidx.lifecycle.viewModelScope
import com.sjani.usnationalparkguide.data.entity.CampEntity
import com.sjani.usnationalparkguide.data.repository.ParkRepository
import kotlinx.coroutines.flow.SharingStarted
import kotlinx.coroutines.flow.stateIn

class CampDetailViewModel(
    repository: ParkRepository,
    campId: String
) : ViewModel() {
    
    val campground = repository.getCampground(campId)
        .stateIn(viewModelScope, SharingStarted.Lazily, null)
}

class CampDetailViewModelFactory(
    private val repository: ParkRepository,
    private val campId: String
) : ViewModelProvider.Factory {
    @Suppress("UNCHECKED_CAST")
    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        if (modelClass.isAssignableFrom(CampDetailViewModel::class.java)) {
            return CampDetailViewModel(repository, campId) as T
        }
        throw IllegalArgumentException("Unknown ViewModel class")
    }
}


